### PR TITLE
New docker worflow

### PR DESCRIPTION
**Pull request by pbench**
New worflow that :

-  create navitia packages 
-  build the dockers from the navitia packages (I imported the contents of hove-io/navitia-docker-compose)
- push the images on dockerhub
- trigger the push of the images on aws

### CODE REVIEW

File: docker/debian8/Dockerfile-kraken [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+# Build libkeepalive https://libkeepalive.sourceforge.net/
+# libkeepalive is a library to keep tcp connection alive. The reason of doing so is that aws LoadBalancer, to which
+# kraken is connected, check the aliveness continuously at regular interval of non-customisable 350s
+# (https://docs.aws.amazon.com/elasticloadbalancing/latest/network/network-load-balancers.html). If nothing happens during
+# this interval, the connection will be dropped. It takes Kraken, by default, 7200s(/proc/sys/net/ipv4/tcp_keepalive_time)
+# to be aware of the disconnection. By using libkeepalive, we can override tcp_keepalive_time,
+# which is not customisable via docker-container.
+
+WORKDIR /tmp/libkeepalive
+COPY navitia/docker/libkeepalive-0.3.tar.gz /tmp/libkeepalive/libkeepalive-0.3.tar.gz
```
Comment (woshilapin) [inline docker/debian8/Dockerfile-kraken]: I believe you're missing this `libkeepalive-0.3.tar.gz` file in the repository. Maybe it'd be a better idea to download directly from Internet ([link](http://prdownloads.sourceforge.net/libkeepalive/libkeepalive-0.3.tar.gz?download))?

File: .github/workflows/build_dockers.yml [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+      run: |
+          echo '{"text":":octopus: Github Actions: build_dockers succeeded. New navitia ${{env.navitia_tag}} image available.' | http --json POST ${{secrets.SLACK_NAVITIA_CORE_TEAM_URL}}
+
+
+    # - name: new release notification
+    #   if: success() && startsWith(github.ref, 'refs/tags/')
+    #   run: |
+    #       version_number=
+    #       echo '{"text":":information_source: Navitia Github Actions: build_navitia_packages_for_release succeded -' ${{env.navitia_tag}} 'navitia docker images are available"}' | http --json POST ${{secrets.SLACK_NAVITIA_CORE_TEAM_URL}}
+    #       echo '{"text":":octopus: Navitia Release: The version' $${{env.navitia_tag}} 'is available.'"}' | http --json POST ${{secrets.SLACK_NAVITIA_URL}}
```
Comment (woshilapin) [inline .github/workflows/build_dockers.yml]: Should this be uncommented?

File: .github/workflows/build_dockers.yml [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        components='jormungandr kraken tyr-beat tyr-worker tyr-web instances-configurator mock-kraken eitri'
+        for component in $components; do
+            echo "*********  Building $component ***************"
+            docker build -t navitia/$component:${{env.navitia_tag}} -f  navitia/docker/${{env.debian_version}}/Dockerfile-${component} .
+
+            # add latest tag if navitia_tag != dev
+            if [[ "${{env.navitia_tag}}" != "dev" ]]; then
+                docker tag navitia/$component:${{env.navitia_tag}}  navitia/$component:latest
+            fi
+        done
```
Comment (woshilapin) [inline .github/workflows/build_dockers.yml:128]: I'm wondering, would it be feasible to create a Github Actions `matrix` to parallelize the build of these components? Might be too much for this PR at the moment though :shrug:

**Review by woshilapin — APPROVED** [review]
I'd say we don't need all the Debian 10 stuff since we have problem with it. How about not keeping it here?

File: .github/workflows/build_dockers.yml (hunk 3 above)
Comment (pbench) [reply .github/workflows/build_dockers.yml:128]: Yeah a bit too much for this PR. These steps are not so long anyway, they mostly just "apt install" debian packages. I don't think it worth the hassle to try to parallelize them

**Review by woshilapin — APPROVED** [review]

**Comment by pbench** [comment]
> I'd say we don't need all the Debian 10 stuff since we have problem with it. How about not keeping it here?

I think (?) these are also useful for debian11 (which want to switch to). Maybe @xlqian can confirm ?

**Comment by pbench** [comment]
I merge this PR to test if everything works. 
I'll do a subsequent PR to clean things up.

**Pull request merged**